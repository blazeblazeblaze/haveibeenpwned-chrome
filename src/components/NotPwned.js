import React from 'react';
import RestartLink from './RestartLink';

const NotPwned = ({onStartOver}) => {
  return (
    <div>
      <p>
        <strong>Happy days!</strong> It looks like you've never been pwned!
      </p>
      <RestartLink onStartOver={onStartOver} />
    </div>
  );
}

export default NotPwned;
