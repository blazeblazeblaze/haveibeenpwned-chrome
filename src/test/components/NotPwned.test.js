import React from 'react';
import ReactDOM from 'react-dom';
import NotPwned from '../../components/NotPwned';
import renderer from 'react-test-renderer';

jest.mock('../../components/RestartLink', () => () => 'RestartLink');

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NotPwned onStartOver={jest.fn } />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('NotPwned renders sad text and links', () => {
  const component = renderer.create(
    <NotPwned onStartOver={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
