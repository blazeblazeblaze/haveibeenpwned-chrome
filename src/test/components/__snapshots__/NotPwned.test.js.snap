// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotPwned renders sad text and links 1`] = `
<div>
  <p>
    <strong>
      Happy days!
    </strong>
     It looks like you've never been pwned!
  </p>
  RestartLink
</div>
`;
